src/transform.js

```diff
@@ -0,0 +1,41 @@
+import { properties as childProperties } from "./children.js";
+import { matches } from "./util.js";
+
+
+/**
+ * Recursively execute a callback on this node and all its children.
+ * If the callback returns a non-undefined value, it will overwrite the node.
+ * This function will not modify the root node of the input AST.
+ * 
+ * @param {object | object[]} node AST node or array of nodes
+ * @param {function(object, string, object?)} transformCallback
+ * @param {object} [o]
+ * @param {string | string[] | function} [o.only] Only walk nodes of this type
+ * @param {string | string[] | function} [o.except] Ignore walking nodes of these types
```

Comment:
Missing `@returns`

Reply:
Do we want an @returns for a void function?

Reply:
Didn't even notice it was void, which showcases why you'd still need to mark it. :) Here are a few ways: https://stackoverflow.com/questions/4759175/how-to-return-void-in-jsdoc

But also, why not just return the result of the callback for the root node? That way authors can still take care of transforming the root node, if that's what they want.

Reply:
Great idea! I will make PR for parents.get then do this

Reply:
What should happen in the case where someone passes in an array of nodes since we updated the spec of this function to take in either an array of nodes or a single object?

Reply:
> What should happen in the case where someone passes in an array of nodes since we updated the spec of this function to take in either an array of nodes or a single object?

Nevermind, I just wrote in the docs that it can return an array of objects

Reply:
I'd expect it to return an array of transformed objects. Is that what you did?

Reply:
Yeah that's what I did

Reply:
@LeaVerou just making sure you saw this, any other feedback? Would like to get this merged so then I can move on to map, then add the sugar for passing in an object or a callback